File: packages/grafana-faro/ios/Plugin/Classes/Options/SetSessionOptions.swift
import Capacitor
import Foundation

@objc public class SetSessionOptions: NSObject {
    private let attributes: [String: String]?
    private let id: String?

    init(call: CAPPluginCall) {
        self.attributes = call.getObject("attributes") as? [String: String]
        self.id = call.getString("id")
    }

    func getAttributes() -> [String: String]? {
        return attributes
    }

    func getId() -> String? {
        return id
    }
}

File: packages/grafana-faro/android/src/main/java/io/capawesome/capacitorjs/plugins/grafanafaro/classes/options/SetSessionOptions.java
package io.capawesome.capacitorjs.plugins.grafanafaro.classes.options;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.grafanafaro.classes.JsonUtils;
import java.util.Map;

public class SetSessionOptions {

    @Nullable
    private final Map<String, String> attributes;

    @Nullable
    private final String id;

    public SetSessionOptions(@NonNull PluginCall call) {
        this.attributes = JsonUtils.toStringMap(call.getObject("attributes"));
        this.id = call.getString("id");
    }

    @Nullable
    public Map<String, String> getAttributes() {
        return attributes;
    }

    @Nullable
    public String getId() {
        return id;
    }
}
